lib/utils.js

```diff
@@ -481,68 +481,37 @@ class Utils {
      * This function reads each section file and wraps it with a markdown header
      * @returns <string> the content of the markdown file as string, warped in a collapsable section.
      */
-    static readCLIMarkdownSectionsAndWrap() {
+    static readCommandSummaryMarkdown() {
         return __awaiter(this, void 0, void 0, function* () {
-            const outputDir = Utils.getJobOutputDirectoryPath();
-            let markdownContent = '';
-            const sectionContents = {};
-            // Read all sections.
-            for (const sectionName of Utils.JOB_SUMMARY_MARKDOWN_SECTIONS_NAMES) {
-                const fullPath = path.join(outputDir, sectionName, 'markdown.md');
-                if ((0, fs_1.existsSync)(fullPath)) {
-                    sectionContents[sectionName] = yield Utils.readSummarySection(fullPath, sectionName);
-                }
-            }
-            // If build info was published, remove generic upload section to avoid duplications with generic modules.
-            if (sectionContents[MarkdownSection.BuildInfo] != '') {
-                sectionContents[MarkdownSection.Upload] = '';
-            }
-            // Append sections in order.
-            for (const sectionName of Utils.JOB_SUMMARY_MARKDOWN_SECTIONS_NAMES) {
-                markdownContent += sectionContents[sectionName] || '';
-            }
+            let markdownContent = yield Utils.readMarkdownContent();
+            // Check if the header can be accessed via the internet to decide if to use the image or the text header
+            this.isSummaryHeaderAccessible = yield this.isHeaderPngAccessible();
+            core.debug('Header image is accessible: ' + this.isSummaryHeaderAccessible);
             return markdownContent ? Utils.wrapContent(markdownContent) : '';
         });
     }
-    static readSummarySection(fullPath, section) {
+    static readMarkdownContent() {
         return __awaiter(this, void 0, void 0, function* () {
-            let content = '';
+            const outputDir = Utils.getJobOutputDirectoryPath();
             try {
-                content = yield fs_1.promises.readFile(fullPath, 'utf-8');
-                return Utils.wrapCollapsableSection(section, content);
+                return yield fs_1.promises.readFile(path.join(outputDir, 'markdown.md'), 'utf-8');
             }
             catch (error) {
-                throw new Error('failed to read section file: ' + fullPath + ' ' + error);
+                core.debug("No markdown file found in the job's output directory.");
```

Comment:
* It is best to add the error message to the log message. The same goes for logging in the `isHeaderPngAccessible` function.
* Since the error can be thrown for a variety of reasons, and not just in case the file doesn't exist, perhaps it would be better to add a specific check for the existence of the file before reading it. That is, if our code actually expects a scenario where the file shouldn't exist. If the code doesn't expect such a scenario, then the logging here should an error log and not a debug one.